core/src/main/java/org/jivesoftware/AccountCreationWizard.java

```diff
@@ -353,7 +360,23 @@ private XMPPConnection getConnection() throws SmackException, IOException, XMPPE
         {
             builder.setHost( localPreferences.getXmppHost() );
         }
-
+        
+        if (securityMode != ConnectionConfiguration.SecurityMode.disabled && !useOldSSL) {
+            builder.setPort(5222);
```

Comment:
Does this take into account that the user can also set another port than 5222? If that's done, we should not hard-code to 5222.